packages/medusa/src/api/middlewares/with-default-sales-channel.ts

```diff
@@ -0,0 +1,28 @@
+import { NextFunction, Request, Response } from "express"
+import { SalesChannelService } from "../../services"
+
+/**
+ * Middleware that includes the default sales channel on the request, if no sales channels present
+ * @param asArray Whether to include the default sales channel id as an array or just a string
+ */
+export function withDefaultSalesChannel(
+  asArray?: boolean
```

Comment:
suggestion(non-blocking): Create a context object to make the middleware more verbose.
```suggestion
  { attachChannelsAsArray?: boolean }
```

Reply:
That was my first approach as well, but then it can't be optional, can it?

Reply:
That was my first approach as well, but then it makes the optionality a bit odd, as you won't be able to just call `withDefaultSalesChannel()`, as it expects 1 arg, even if the only key in it is optional.